pennylane/measurements/measurements.py

```diff
@@ -129,8 +134,83 @@ class MeasurementProcess(ABC):
 
     # pylint:disable=too-many-instance-attributes
 
+    _obs_primitive: Optional["jax.core.Primitive"] = None
+    _wires_primitive: Optional["jax.core.Primitive"] = None
+    _mcm_primitive: Optional["jax.core.Primitive"] = None
+
     def __init_subclass__(cls, **_):
         register_pytree(cls, cls._flatten, cls._unflatten)
+        name = getattr(cls.return_type, "value", cls.__name__)
+        cls._wires_primitive = qml.capture.create_measurement_wires_primitive(cls, name=name)
+        cls._obs_primitive = qml.capture.create_measurement_obs_primitive(cls, name=name)
+        cls._mcm_primitive = qml.capture.create_measurement_mcm_primitive(cls, name=name)
+
+    @classmethod
+    def _primitive_bind_call(cls, obs=None, wires=None, eigvals=None, id=None, **kwargs):
+        """Called instead of ``type.__call__`` if ``qml.capture.enabled()``.
+
+        Measurements have three "modes":
+
+        1) Wires or wires + eigvals
+        2) Observable
+        3) Mid circuit measurements
+
+        Not all measurements support all three modes. For example, ``VNEntropyMP`` does not
+        allow being specified via an observable. But we handle the generic case here.
+
+        """
+        if cls._obs_primitive is None or cls._wires_primitive is None or cls._mcm_primitive is None:
```

Comment:
I'm not quite sure I understand the `or` relationship here :thinking: Could you elaborate what's behind that? Thanks!

Reply:
Potentially we could just check one.  

This is just kinda a safety clause to not use capture if any of the primitives wasn't initialized correctly. 

But I guess if capture is enabled, then jax is installed. And if jax is installed, then the primitives should be set.

Reply:
Ah, got it now, thanks!
I don't mind either way, I think it's  a good intuition to go for safety here, but one really has to meddle with the internals to get this one to not be equivalent with just `if cls._obs_primitive is None:`